junit5/core/src/main/java/org/jboss/arquillian/junit5/JUnitJupiterTestClassLifecycleManager.java

```diff
@@ -1,47 +1,96 @@
 package org.jboss.arquillian.junit5;
 
+import org.jboss.arquillian.test.spi.TestRunnerAdaptor;
+import org.jboss.arquillian.test.spi.TestRunnerAdaptorBuilder;
+import org.junit.jupiter.api.extension.ExtensionContext;
+
 import java.lang.reflect.InvocationTargetException;
 import java.lang.reflect.Method;
 import java.util.Optional;
 
-import org.jboss.arquillian.test.spi.TestRunnerAdaptor;
-import org.junit.jupiter.api.extension.ExtensionContext;
-
-public class JUnitJupiterTestClassLifecycleManager extends ArquillianTestClassLifecycleManager {
+public class JUnitJupiterTestClassLifecycleManager implements ExtensionContext.Store.CloseableResource {
     private static final String NAMESPACE_KEY = "arquillianNamespace";
 
-    private static final String ADAPTOR_KEY = "testRunnerAdaptor";
-
-    private static final String SUITE_FINALIZER_KEY = "suiteFinalizer";
+    private static final String MANAGER_KEY = "testRunnerManager";
 
     private static final String INTERCEPTED_TEMPLATE_NAMESPACE_KEY = "interceptedTestTemplates";
 
     private static final String RESULT_NAMESPACE_KEY = "results";
 
-    private ExtensionContext.Store store;
+    private final ExtensionContext.Store templateStore;
+
+    private final ExtensionContext.Store resultStore;
 
-    private ExtensionContext.Store templateStore;
+    private TestRunnerAdaptor adaptor;
 
-    private ExtensionContext.Store resultStore;
+    private Throwable caughtInitializationException;
 
     JUnitJupiterTestClassLifecycleManager(ExtensionContext context) {
-        store = context.getRoot().getStore(ExtensionContext.Namespace.create(NAMESPACE_KEY));
-        templateStore = context.getStore(ExtensionContext.Namespace.create(NAMESPACE_KEY, INTERCEPTED_TEMPLATE_NAMESPACE_KEY));
-        resultStore = context.getStore(ExtensionContext.Namespace.create(NAMESPACE_KEY, RESULT_NAMESPACE_KEY));
+        templateStore = context.getRoot().getStore(ExtensionContext.Namespace.create(NAMESPACE_KEY, INTERCEPTED_TEMPLATE_NAMESPACE_KEY));
```

Comment:
Its far from optimal to keep all results/templates in root context.. (with lot of tests it may be problem)
It should go back to context, but we need to switch context for each test or we may loose data..

It requires bot of refactoring but its fairly easy..